classes/module/Module.php

```diff
@@ -221,6 +221,8 @@ abstract class ModuleCore implements ModuleInterface
 
     public static $hosted_modules_blacklist = ['autoupgrade'];
 
+    public static $customer_dependent_module_list = ['ps_featuredproducts', 'ps_bestsellers', 'ps_specials', 'ps_crossselling', 'ps_newproducts'];
+
```

Comment:
Instead of hardening a list of modules,  add in core something look :  

```php
public function dependOnCustomer(){
    return false; // By default
}
```
and each module must declare whether or not it depends on the customer.

```php
class MyModule extends Module {
   public function dependOnCustomer(): bool{
      return true;
   }
}
```

And you can check  :  

```php
 if ($this->dependOnCustomer() && !empty($this->context->customer->id)) {
            $cache_array[] = (int) $this->context->customer->id;
 }
```

Reply:
But @davidglezz  [proposal](https://github.com/PrestaShop/PrestaShop/pull/22859#issuecomment-761064874) seems better to me !

Reply:
Modules like ps_banner or ps_socialfollow don(t depends on customer group, so the method should be dependOn() with a list of several dependances

Reply:
@PululuK @jf-viguier 

I'd go for your solution @PululuK but instead of having it in method I'd go for just having a public property, then all we need to do is to have:

`$this->cache_by_customer = true;` or "cache_depends_on_customer" or some other name